core/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -1385,12 +1393,66 @@ enum SafeRexVisitor implements RexVisitor<Boolean> {
     }
 
     @Override public Boolean visitCall(RexCall call) {
-      if (!safeOps.contains(call.getKind())) {
+      SqlKind sqlKind = call.getKind();
+      switch (sqlKind) {
+      case DIVIDE:
+      case MOD:
+        List<RexNode> operands = call.getOperands();
+        // For example:
+        // a/b ==>
+        // if a is not safe then a/b is not safe
```

Comment:
I am not sure I understand what's going on here.
But dividing by -1 is not safe, since MIN_INT / -1 causes an overflow

Reply:
I consider 0 as unsafe, while other numbers are safe. `MIN_INT / -1`I also have doubts about this, but I will consider it safe here because it actually depends on the return type of the operator.

Reply:
In PostgreSQL, it throws `integer out of range`. So I will revert this change.